### PR TITLE
Change of License from LGPL to **MPL 2.0**

### LINKED ISSUE #7: LGPL causes issues in combination with FMUs
I noticed that your library is licensed under LGPL now there is an issue here since LGPL does not allow static linking with other non-lgpl code. This basically means you can not use your library in combination with any other library (including the MSL btw) if you generate an FMU that you like to distribute. There might be other cases. So you might wanna think of re-licensing under MPL which basically allows static linking but is otherwise similar to lgpl.  See also the section on this in our [impact paper at the last Modelica conference](https://github.com/impact/impact/blob/master/resources/docs/modelica2015/paper/impact.md#which-license-is-best-for-your-library).

### LINKED ISSUE #62: Move copyright notice into "revisions" section?
I noticed that you have the copyright notice present in all models in the "info" layer of the documentation. You might consider using the "revisions" layer instead for this purpose. This way it's easier to distinguish between real info and just default copyright and authors stamp. Quite a lot libraries do it this way but is of course not a must.

A step further would be to not have the copyright notice in all models in the development version but a simple place holder (e.g., `<!-- DISCLAIMER -->` which is legal HTML and will not cause HTML checker to complain :-) ) that gets replaced with an updated copyright notice before release. That way your source code does not have lots of duplicate code that needs to be updated with every release. Just an idea.

### PR DESCRIPTION
**Pull request by MaximeBaudette**
The copyright statement and license info is now moves in the revision part starting with 

 

to identify the section

closes #7, closes #62

### CODE REVIEW

**Comment by MaximeBaudette** [comment]
@geofjamg 
I'd prefer someone else takes a look before integrating the changes, that's why I went for the pull request instead of directly pushing to the repository. 

When you merge it, I'll have to do the following:
- Update readme file
- Update wiki 
- search and update any remaining reference to LGPL

**Comment by dietmarw** [comment]
There is indeed a problem with the changes. You added:

```
revisions="<!--DISCLAIMER-->
<html>
....
</html>
```

The problem is that your HTML comment element `<!--DISCLAIMER-->` now sits OUTSIDE the HTML part of the revision. Which means it contains now a text part and an HTML part. This is bound to lead to problems for the display.

Better change it to:

```
revisions="<html>
<!--DISCLAIMER-->
....
</html>
```

I.e., start the revision string right away with the `<html>` identifier.

**Comment by dietmarw** [comment]
Qed:

```
model Unnamed
annotation (Documentation(revisions="<!--DISCLAIMER-->
<html>
Foo 
</html>"));
end Unnamed;
```

image

vs the correct version:

```
model Unnamed
annotation (Documentation(revisions="<html>
<!--DISCLAIMER-->
Foo
</html>"));
end Unnamed;
```

image

File: copyrightStatement.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!--DISCLAIMER-->
+<html>
+<p>Copyright 2015 RTE (France), SmarTS Lab (Sweden), AIA (Spain) and DTU (Denmark)</p>
+<p>?- ?RTE: http://www.rte-france.com/</p>
```
Comment (dietmarw) [inline copyrightStatement.txt]: What are those `?` artefacts from copy paste?

File: iPSL/Connectors/package.mo [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+
+
+
+
+
+annotation (Documentation(info="", revisions="<!--DISCLAIMER-->
+<html>
+<p>Copyright 2015 RTE (France), SmarTS Lab (Sweden), AIA (Spain) and DTU (Denmark)</p>
+<p>​- ​RTE: http://www.rte-france.com/</p>
+<p>- SmarTS Lab, research group at KTH: https://www.kth.se/en</p>
```
Comment (dietmarw) [inline iPSL/Connectors/package.mo]: Wouldn't it be better to make URLs also clickable by embedding them in an `<a>` environment? This is after all HTML ;-)

File: iPSL/Connectors/package.mo [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+<p>​- ​AIA: http://www.aia.es/en/energy/</p>
+<p>​- ​DTU: http://www.dtu.dk/english</p>
+<p>The authors can be contacted by email: info@itesla-ipsl.org </p>
```
Comment (dietmarw) [inline iPSL/Connectors/package.mo]: Same for emails: `<a href=\"mailto:info@itesla-ipsl.org\">info@itesla-ipsl.org</a>`

File: iPSL/Connectors/package.mo [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
-<p><span style=\"font-family: MS Shell Dlg 2;\">You should have received a copy of the GNU Lesser General Public License along with the iPSL. If not, see &LT;http://www.gnu.org/licenses/&GT;.</span></p>
+
+
+
+
+
+annotation (Documentation(info="", revisions="<!--DISCLAIMER-->
+<html>
+<p>Copyright 2015 RTE (France), SmarTS Lab (Sweden), AIA (Spain) and DTU (Denmark)</p>
+<p>​- ​RTE: http://www.rte-france.com/</p>
```
Comment (dietmarw) [inline iPSL/Connectors/package.mo]: make URLs consistent either with or without trailing `/`. But better don't mix.

File: iPSL/Connectors/package.mo [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
-<p><span style=\"font-family: MS Shell Dlg 2;\">This package is part of the iTesla Power System Library (&QUOT;iPSL&QUOT;) .</span></p>
-<p><span style=\"font-family: MS Shell Dlg 2;\">The iPSL is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.</span></p>
-<p><span style=\"font-family: MS Shell Dlg 2;\">The iPSL is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.</span></p>
-<p><span style=\"font-family: MS Shell Dlg 2;\">You should have received a copy of the GNU Lesser General Public License along with the iPSL. If not, see &LT;http://www.gnu.org/licenses/&GT;.</span></p>
+
+
+
+
+
+annotation (Documentation(info="", revisions="<!--DISCLAIMER-->
```
Comment (dietmarw) [inline iPSL/Connectors/package.mo]: better remove empty info annotations. Makes it easier later on to scan for models which are still missing documentation.

File: iPSL/Electrical/Banks/CSVGN1.mo [hunk 6]
```diff
@@ -121,19 +121,16 @@ equation
...
-<p><span style=\"font-family: MS Shell Dlg 2;\">This package is part of the iTesla Power System Library (&QUOT;iPSL&QUOT;) .</span></p>
-<p><span style=\"font-family: MS Shell Dlg 2;\">The iPSL is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.</span></p>
-<p><span style=\"font-family: MS Shell Dlg 2;\">The iPSL is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.</span></p>
-<p><span style=\"font-family: MS Shell Dlg 2;\">You should have received a copy of the GNU Lesser General Public License along with the iPSL. If not, see &LT;http://www.gnu.org/licenses/&GT;.</span></p>
+    Documentation(info="", revisions="<!--DISCLAIMER-->
+<html>
+<p>Copyright 2015 RTE (France), SmarTS Lab (Sweden), AIA (Spain) and DTU (Denmark)</p>
```
Comment (dietmarw) [inline iPSL/Electrical/Banks/CSVGN1.mo]: use the power of HTML for lists: 

```
<ul>
 <li>​RTE: http://www.rte-france.com</li>
 <li> ... </li>
</ul>
```

File: copyrightStatement.txt (hunk 1 above)
Comment (MaximeBaudette) [inline copyrightStatement.txt]: right... I got a prompt of character conversion... but didn't pay attention to it, cause these stuff usually happens on windows machines. 
Glad you found that error :-)

File: iPSL/Connectors/package.mo (hunk 4 above)
Comment (MaximeBaudette) [inline iPSL/Connectors/package.mo]: that I can fix !

File: iPSL/Connectors/package.mo (hunk 3 above)
Comment (MaximeBaudette) [inline iPSL/Connectors/package.mo]: sure

File: iPSL/Connectors/package.mo (hunk 2 above)
Comment (MaximeBaudette) [inline iPSL/Connectors/package.mo]: sure

**Comment by MaximeBaudette** [comment]
@geofjamg 
ok... so after @dietmarw 's comments I'll do some changes... but right now I wanna sleep so I'll take it tomorrow morning. 

In the mean time, NO Merging.

File: iPSL/Connectors/package.mo (hunk 5 above)
Comment (MaximeBaudette) [inline iPSL/Connectors/package.mo]: I'll try to figure something out

File: iPSL/Electrical/Banks/CSVGN1.mo (hunk 6 above)
Comment (MaximeBaudette) [inline iPSL/Electrical/Banks/CSVGN1.mo]: sure. 
Note: I'm not a master at HTML... and going back and forth between the Dymola viewer and editor has created problems for me (insertion of non wanted code)

**Comment by MaximeBaudette** [comment]
@dietmarw 
I tried 

```
revisions="<html>
<!--DISCLAIMER-->
....
</html>
```

But Dymola removed the  , that's why I didn't do it... but now I tested again, and it's just that if you happen to edit the HTML via the WYSIWYG editor, then it removes it, doesn't matter where it is placed... 
So in short, I'll move it as you suggest.

*\* Bonus Question**: Do you know if there is a way of blocking the HTML editor of Dymola?

**Comment by dietmarw** [comment]
I thought you wanted to go to sleep? ;-) Anyway I can confirm that Dymola removes the HTML comment. I've reported a ticket on this. My standard advice stands. **Hands off of Dymola's WYSIWYG editor!**

By editing all those files I guess you might have appreciated my more sophisticated solution (which would probably have cost you less time). I.e., a script that replaces the place holder `<!--DISCLAIMER-->` by the actual disclaimer. If interested there are a whole set of nice helper scripts which can be adapted to your need in https://github.com/lbl-srg/modelica-buildings/tree/master/bin

Anyway since you do **not** yet use the place holder then you do not need to have it present. My solution I suggested was have **either** (and preferable) the disclaimer _placeholder_ (and only that one) in the source and add the actual text via running a script for the released version only **or** add the disclaimer in all those places manually. The latter you've seen is quite cumbersome and in addition also prone to have duplicate errors and differences and is a nightmare to keep up to date.

Having said that, I have actually a much better idea which also makes use of Modelica's inheritance features and is easy to maintain :-) Give me an hour and I'll send a PR against master and let me know what you think.

**Comment by MaximeBaudette** [comment]
@dietmarw 
Actually I took inspiration of this this place holder by keeping it even when the copyright info is present.
Since I usually do as follows when updating the license info:
- Save the lib as one file
- Open it in a regular text editor (Atom)
- Use search and replace to update the license info
- back to Dymola and saving as a broken down lib 

Given this workflow I can easily keep the placeholder to identify where to update the license.
And also... now I have done the job of putting it everywhere (which was cumbersome), but we don't need to do it anymore.

**Comment by dietmarw** [comment]
Well you need to update it at least once every year. BTW we have 2016 now :smile:

**Comment by MaximeBaudette** [comment]
yes, I was wondering about that 2016... :smile: 
I'll add it as 2015-2016

**Comment by dietmarw** [comment]
Well wait for my PR and maybe you like that solution better.

**Comment by MaximeBaudette** [comment]
No problem ! 
I'm just preparing our "for the time being" alternative. 
Then I'll email the other collaborators to discuss with them. It will be much easier to discuss the two alternatives if I can show how things are done in the modelica code. And also I can explain what are pros and cons of both alternatives. 
So when you send your PR, could quickly describe what you see as pros and cons, so I can grasp it

**Comment by MaximeBaudette** [comment]
@dietmarw 
Ok, now I took all your comments into account... Took me much longer than I thought as I ran into an encoding issue and was missing the escaping characters. 
I'm very curious about your PR to come :smile:

**Pull request merged**